src/NuGet.Jobs.RegistrationComparer/HiveComparer.cs

```diff
@@ -97,67 +103,71 @@ public class HiveComparer
                     .Data
                     .Items
                     .Where(p => p.Items == null)
-                    .Select(p => p.Url)
+                    .Select(p => new { x.Hive, p.Url })
                     .ToList())
                 .ToList();
-            var leafUrlGroups = baseUrls
-                .Select(x => versions.Select(v => $"{x}{id}/{v}.json").ToList())
+            var leafUrlGroups = hives
+                .Select(x => versions
+                    .Select(v => $"{x.BaseUrl}{id}/{v}.json")
+                    .Select(u => new { Hive = x, Url = u })
+                    .ToList())
                 .ToList();
 
-            var urls = new ConcurrentBag<string>(pageUrlGroups
+            var pairs = new ConcurrentBag<KeyValuePair<string, HiveConfiguration>>(pageUrlGroups
                 .SelectMany(x => x)
-                .Concat(leafUrlGroups.SelectMany(x => x)));
+                .Concat(leafUrlGroups.SelectMany(x => x))
+                .Select(x => new KeyValuePair<string, HiveConfiguration>(x.Url, x.Hive)));
```

Comment:
`KeyValuePair.Create(x.Url, x.Hive)` would be a bit more compact.
What is `Url` here, by the way?

Reply:
This is the official (not blob storage) URL of a registration leaf that needs to be downloaded and validated.

Reply:
The URL is there so we can parallelize the downloads of many leaves and pages.

I'll use the `KeyValuePair.Create` trick.